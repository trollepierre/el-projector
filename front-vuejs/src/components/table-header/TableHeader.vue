<template>
  <thead>
    <th class="next">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'Next'"
        @click="sortByNext"
      />
    </th>
    <th class="first-name">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'FirstName'"
        @click="sortByFirstName"
      />
    </th>
    <th class="last-name">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'LastName'"
        @click="sortByLastName"
      />
    </th>
    <th class="points">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'Points'"
        @click="sortByPoints"
      />
    </th>
    <th class="adjusted-points">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'Adjusted Points'"
        @click="sortByPoints"
      />
    </th>
    <th class="last-date">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'LastDate'"
        @click="sortByLastDate"
      />
    </th>
    <th class="impact">
      Impact
    </th>
    <th class="birth-date">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'BirthDate'"
        @click="sortBirthday"
      />
    </th>
    <th class="origin">
      <AppButton
        :background-color="'cyan'"
        :color="'black'"
        :text="'Origin'"
        @click="sortByOrigin"
      />
    </th>
    <th class="action">
      Action
    </th>
  </thead>
</template>
<script>
  import AppButton from '../common/app-button/AppButton.vue'

  export default {
    name: 'TableHeader',
    components: { AppButton },
    methods: {
      sortBirthday() {
        this.$emit('sortBirthday')
      },
      sortByFirstName() {
        this.$emit('sortByFirstName')
      },
      sortByLastDate() {
        this.$emit('sortByLastDate')
      },
      sortByLastName() {
        this.$emit('sortByLastName')
      },
      sortByNext() {
        this.$emit('sortByNext')
      },
      sortByOrigin() {
        this.$emit('sortByOrigin')
      },
      sortByPoints() {
        this.$emit('sortByPoints')
      },
    },
  }
</script>
<style scoped>

  thead {
    display: flex;
    height: 40px;
  }

  thead {
    width: 100%;
  }

  th {
    flex: 1;
    align-items: center;
    justify-content: center;
    display: flex;
  }

  .impact, .origin {
    flex-grow: 2;
  }

</style>
