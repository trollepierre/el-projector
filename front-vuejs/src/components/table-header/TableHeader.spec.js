import TableHeader from './TableHeader.vue'

describe('TableHeader.vue', () => {
  it('should match snapshot', () => {
    // When
    const wrapper = shallowMount(TableHeader)

    // Then
    expect(wrapper.element).toMatchSnapshot()
  })
})
