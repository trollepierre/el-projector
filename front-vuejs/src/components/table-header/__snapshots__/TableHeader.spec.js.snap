// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader.vue should match snapshot 1`] = `
<thead>
  <th
    class="next"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="Next"
    />
  </th>
   
  <th
    class="first-name"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="FirstName"
    />
  </th>
   
  <th
    class="last-name"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="LastName"
    />
  </th>
   
  <th
    class="points"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="Points"
    />
  </th>
   
  <th
    class="adjusted-points"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="Adjusted Points"
    />
  </th>
   
  <th
    class="last-date"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="LastDate"
    />
  </th>
   
  <th
    class="impact"
  >
    
    Impact
  
  </th>
   
  <th
    class="birth-date"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="BirthDate"
    />
  </th>
   
  <th
    class="origin"
  >
    <appbutton-stub
      backgroundcolor="cyan"
      color="black"
      text="Origin"
    />
  </th>
   
  <th
    class="action"
  >
    
    Action
  
  </th>
</thead>
`;
